src/ObjectMakr.ts

```diff
@@ -226,15 +225,17 @@ export class ObjectMakr implements IObjectMakr {
         // For each name in the current object:
         for (let name in base) {
             if (base.hasOwnProperty(name)) {
-                this.functions[name] = new Function() as IClassFunction;
+                if (!this.functions[name]) {
+                    this.functions[name] = new Function() as IClassFunction;
 
-                // This sets the Function as inheriting from the parent
-                this.functions[name].prototype = new parent();
-                this.functions[name].prototype.constructor = this.functions[name];
+                    // This sets the Function as inheriting from the parent
+                    this.functions[name].prototype = new parent();
+                    this.functions[name].prototype.constructor = this.functions[name];
+                }
 
                 // Add each property from properties to the Function prototype
                 for (let ref in this.properties[name]) {
-                    if (this.properties[name].hasOwnProperty(ref)) {
+                    if (this.properties[name].hasOwnProperty(ref) && !this.functions[name].prototype[ref]) {
```

Comment:
You commented before on passing in some functions but not others, so in case some properties were not defined, they would be here. Unless the functions that were passed in were complete, then yes there would be no reason for this.